tests/unit/CoreHeadingTest.php

```diff
@@ -0,0 +1,486 @@
+				'style'           => wp_json_encode([
+					'color' => [
+						'background' => '#cf2e2e',
+						'text'       => '#ffffff',
+					],
+				]),
+				'textAlign'       => 'right',
+				'textColor'       => null, // @todo: This is not returning correctly.
+			],
+			$attributes
+		);
+	}
+
+	public function test_retrieve_core_heading_with_font_and_anchor() {
+		$block_content = '
+			<!-- wp:heading {"anchor":"custom-id","style":{"typography":{"fontFamily":"Arial","fontSize":"32px"}}} -->
+			<h2 class="wp-block-heading" id="custom-id" style="font-family:Arial;font-size:32px">Custom Font Heading</h2>
+			<!-- /wp:heading -->
+		';
+
+		wp_update_post(
+			[
+				'ID'           => $this->post_id,
+				'post_content' => $block_content,
+			]
+		);
+
+		$actual = graphql(
+			[
+				'query'     => $this->query(),
+				'variables' => [ 'id' => $this->post_id ],
+			]
+		);
+
+		$block      = $actual['data']['post']['editorBlocks'][0];
+		$attributes = $block['attributes'];
+
+		$this->assertEquals(
+			[
+				'align'           => null,
+				'anchor'          => 'custom-id', // Previously untested
+				'backgroundColor' => null,
+				'className'       => null,
+				'content'         => 'Custom Font Heading',
+				'cssClassName'    => 'wp-block-heading',
+				'fontFamily' => null, // @todo: This is not returning correctly.
+				'fontSize' => null, // @todo: This is not returning correctly.
+				'gradient'        => null,
+				'level'           => 2.0,
+				'lock'            => null,
+				'placeholder'     => null,
+				'style'           => wp_json_encode([
+					'typography' => [
+						'fontFamily' => 'Arial', // Previously untested
+						'fontSize'   => '32px', // Previously untested
+					],
+				]),
+				'textAlign'       => null,
+				'textColor'       => null,
+			],
+			$attributes
+		);
+	}
+
+	public function test_retrieve_core_heading_with_gradient() {
+		$block_content = '
+			<!-- wp:heading {"style":{"color":{"gradient":"linear-gradient(135deg,rgb(6,147,227) 0%,rgb(155,81,224) 100%)"}}} -->
+			<h2 class="wp-block-heading has-background" style="background:linear-gradient(135deg,rgb(6,147,227) 0%,rgb(155,81,224) 100%)">Gradient Heading</h2>
+			<!-- /wp:heading -->
+		';
+
+		wp_update_post(
+			[
+				'ID'           => $this->post_id,
+				'post_content' => $block_content,
+			]
+		);
+
+		$actual = graphql(
+			[
+				'query'     => $this->query(),
+				'variables' => [ 'id' => $this->post_id ],
+			]
+		);
+
+		$block      = $actual['data']['post']['editorBlocks'][0];
+		$attributes = $block['attributes'];
+
+		$this->assertEquals(
+			[
+				'align'           => null,
+				'anchor'          => null,
+				'backgroundColor' => null,
+				'className'       => null,
+				'content'         => 'Gradient Heading',
+				'cssClassName'    => 'wp-block-heading has-background',
+				'fontFamily'      => null,
+				'fontSize'        => null,
+				'gradient' => null, // @todo: This is not returning correctly.
+				'level'           => 2.0,
```

Comment:
It _both_ - and be specific:

```suggestion
				'gradient'        => null, // Previously untested. @todo: This only gets returned in `style.color.gradient`.
```